### PR TITLE
component documentation for frame, system and lidar

### PR DESCRIPTION
**Pull request by adamdbrw**
## Change summary

Component documentation for ROS 2 components.

### Submission Checklist:

* [X] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [X] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [X] **Consistency** - Does the content consistently follow the [Style Guide](https://o3de.org/docs/contributing/to-docs/style-guide/quick-reference)?
* [X] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

File: content/docs/user-guide/components/reference/ros2/core/_index.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+title: ROS 2 Core Components
+linktitle: Core
+description: Core components for Robot Operating System (ROS 2) in Open 3D Engine (O3DE).
+---
+
+There are following ROS 2 Gem core components:
```
Comment (arturkamieniecki) [inline content/docs/user-guide/components/reference/ros2/core/_index.md:7]:
```suggestion
The following pages describe the ROS 2 Gem core components:
```

File: content/docs/user-guide/components/reference/ros2/core/ros2-system-component.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ROS 2 System Component handles several singleton-like behaviors of the simulation. 
+You can make use of its **Node** to conveniently create publishers and subscribers.
+When creating or updating ROS messages, you can use it to get current ROS timestamp from the simulation clock. 
+It is also internally used to publish static and dynamic transforms which are computed through **ROS 2 Frame** components.
+
+Note that the simulation ROS node accessible through component's API is there as a convenience, and you can create your own nodes and executors if you wish.
+
+## ROS2RequestsBus
+
+The `ROS2RequestBus` is an API system bus that is intended for both internal ROS 2 Gem components and external Gems.
```
Comment (arturkamieniecki) [inline content/docs/user-guide/components/reference/ros2/core/ros2-system-component.md:33]:
```suggestion
The `ROS2RequestBus` alongside the `ROS2Interface` is an API system bus and interface intended for both internal ROS 2 Gem components and external Gems.
```

File: content/docs/user-guide/components/reference/ros2/core/ros2-system-component.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+## Usage
+
+ROS 2 System Component handles several singleton-like behaviors of the simulation. 
+You can make use of its **Node** to conveniently create publishers and subscribers.
+When creating or updating ROS messages, you can use it to get current ROS timestamp from the simulation clock. 
+It is also internally used to publish static and dynamic transforms which are computed through **ROS 2 Frame** components.
+
+Note that the simulation ROS node accessible through component's API is there as a convenience, and you can create your own nodes and executors if you wish.
+
+## ROS2RequestsBus
```
Comment (arturkamieniecki) [inline content/docs/user-guide/components/reference/ros2/core/ros2-system-component.md:31]:
```suggestion
## ROS2RequestsBus and ROS2Interface
```

File: content/docs/user-guide/components/reference/ros2/sensors/_index.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+---
+title: ROS 2 Components
+linktitle: ROS 2
+description: Components for robot control with Robot Operating System (ROS 2) in Open 3D Engine (O3DE).
+---
+
+Refer to the [Robotics](/docs/user-guide/interactivity/robotics) documentation to learn more about robotics simulation features of O3DE.
+
+ROS 2 Gem components can be divided into the following categories:
+-  [Lidar Sensor component](ros2-lidar-sensor.md)
```
Comment (arturkamieniecki) [inline content/docs/user-guide/components/reference/ros2/sensors/_index.md:10]: A reference to the sensor-configuration is missing.

**Comment by arturkamieniecki** [comment]
I've also updated the ROS2Frame to reflect some the new ROS2FrameEditorComponent in https://github.com/o3de/o3de.org/pull/2516.

**Comment by michalpelka** [comment]
Please use the development build with DPE enabled for screenshots.

**Review by michalpelka — APPROVED** [review]

File: content/docs/user-guide/components/reference/ros2/_index.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+---
+title: ROS 2 Components
+linktitle: ROS 2
+description: Using Robot Operating System (ROS 2) components in Open 3D Engine (O3DE).
+---
+
+Refer to the [Robotics](/docs/user-guide/interactivity/robotics) documentation to learn more about robotics simulation features of O3DE.
+
+ROS 2 Gem components can be divided into the following categories:
+-  [Core components](core), essential to running simulations independent on type of robot.
```
Comment (moudgils) [inline content/docs/user-guide/components/reference/ros2/_index.md:10]: ...essential to running simulations independent **of the** type of robot.

**Review by moudgils — APPROVED** [review]

File: content/docs/user-guide/components/reference/ros2/_index.md (hunk 5 above)
Comment (michalpelka) [reply content/docs/user-guide/components/reference/ros2/_index.md:10]: Thanks, will apply!

File: content/docs/user-guide/components/reference/ros2/core/_index.md (hunk 1 above)
Comment (michalpelka) [reply content/docs/user-guide/components/reference/ros2/core/_index.md:7]: Thanks!

**Comment by michalpelka** [comment]
Thank you @moudgils and @arturkamieniecki for review. 
Please @arturkamieniecki give approve if you are happy with update.

**Review by arturkamieniecki — APPROVED** [review]

**Pull request merged**